cogs/roles.py

```diff
@@ -0,0 +1,163 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Canary is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Canary. If not, see <https://www.gnu.org/licenses/>.
+
+# discord-py requirements
+import discord
+from discord import utils
+from discord.ext import commands
+
+import codecs
+import configparser
+from .utils.paginator import Pages
+
+
+class Roles(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.roles = self.bot.config.roles
+
+    async def toggleRole(self, ctx, transaction, requestedRole, category):
+        """
+        Assigs a single role to a user with no checks from a category of roles
+        """
+        member = ctx.message.author
+        roles = self.roles.get(category).split(", ")
+
+        # normalize user input to title case
+        requestedRole = requestedRole.title()
+
+        if (transaction == "add") and (requestedRole in roles):
+            role = discord.utils.get(ctx.guild.roles, name=requestedRole)
+            await member.add_roles(role, reason="Self Requested")
+        elif (transaction == "remove") and (requestedRole in roles):
+            role = discord.utils.get(ctx.guild.roles, name=requestedRole)
+            await member.remove_roles(role, reason="Self Requested")
+        else:
+            await ctx.send("Error, that role doesn't exist in this category")
+        return
+
+    async def toggleRestrictedRole(self, ctx, transaction, requestedRole,
+                                   category):
+        """
+        Assigs a single role to a user with no checks from a category of roles
+        """
+        member = ctx.message.author
+        roles = self.roles.get(category).split(", ")
+
+        # normalize user input to title case
+        requestedRole = requestedRole.title()
+
+        if (transaction == "add") and (requestedRole in roles):
+            role = discord.utils.get(ctx.guild.roles, name=requestedRole)
+            # Allow only a single faculty role per user.
+            for category_roles in roles:
+                old_role = discord.utils.get(ctx.guild.roles,
+                                             name=category_roles)
+                if old_role is not None:
+                    await member.remove_roles(old_role,
+                                              reason="Self Requested")
+            await member.add_roles(role, reason="Self Requested")
+        elif (transaction == "remove") and (requestedRole in roles):
+            role = discord.utils.get(ctx.guild.roles, name=requestedRole)
+            await member.remove_roles(role, reason="Self Requested")
+        else:
```

Comment:
maybe this should early-return instead of being a final else
`if requestedRole not in roles`
different one for `transaction not in ("add", "remove")` or whatever for a different err msg like what val's talking abt

then we can move 
```python
role = discord.utils.get(ctx.guild.roles, name=requestedRole)
```
out of the blocks so that line isnt duplicated twice